### DISCUSSION TITLE
Identifying location of secrets detected by Push Protection

### DISCUSSION THREAD

**Discussion by mikeclayton** [community/community#30780] [Code Security]
Q. Is there a way to determine which commit / file / line is being detected as containing secrets when Push Protection is enabled, and / or work out which rule or pattern it's violating *without* simply bypassing secret detection and then seeing what gets reported later?

**Background**

I'm currently getting an error when trying to push some commits to a feature branch on a repository that has Secret Scanning / Push Protection enabled.

However, the error message doesn't give me any clues about exactly *what* is being detected as a secret:

```
POST git-receive-pack (37663 bytes)
remote: error GH009: Secrets detected! This push failed. 
...
...
Pushing to https://github.com/<myOrg>/<myRepo>.git
To https://github.com/<myOrg>/<myRepo>.git
 ! [remote rejected] <local_branch> -> <remote_branch> (push declined due to a detected secret)
error: failed to push some refs to 'https://github.com/<myOrg>/<myRepo>.git'
```

I've searched through the local commits to see if I can work out what's being flagged, but I've drawn a blank. We've got some custom patterns defined, so it's entirely possible those are misbehaving, but without knowing what is being detected I can't really see what I need to do to fix it.

What's frustrating is that Push Protection obviously thinks it's *found* a secret, so it would be good if the location could be included in the error message...

Note: I've also tried visiting the magic url given to bypass Push protection for the secret, but that doesn't give any further details either - it just asks if you want to allow "this secret" without saying what (or where) "this secret" is:

![image](https://user-images.githubusercontent.com/1193763/186655001-e4d1f430-df5e-4981-9ab9-060260627bc9.png)

Any pointers would be appreciated...

**Comment by 15MariamS** [comment]
Hi @mikeclayton, 

Apologies that you ran into this experience! The error message _should_ have details on where the secret was found so that you can take proper remediation action to remove from history.

Can you answer some questions please?
1. Which IDE are you using?
2. Have you used other IDEs and experienced this problem, too?
3. Are you still running into this problem, or did you find any work arounds?

<p align="center"><img width="500" alt="blocked secret" src="https://user-images.githubusercontent.com/81782111/197921887-564d0aee-2004-4160-ac63-639461ebd9f4.png"></p>

**Reply by sergendestroy** [reply]
I have this issue, but I already deleted the access key. Yet, I still get the error, even if I reset the commit
